### PR TITLE
feat(react-components): Move global camera controls to Reveal context

### PR DESCRIPTION
**Pull request by haakonflatval-cognite**
#### Type of change

Feat  

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/

## Description :pencil:

This moves the global camera controls to props in RevealContext. Earlier, they would be given to a component that had been created from the `withCameraStateControl`. This makes it very abritrary where in the code this camera control happens, and can be controlled from multiple places in the same page, which may cause unintended behavior. Now, the controls (a getter and setter) of the camera state are given as props directly to RevealContext, making it more obvious where this global state is controlled. Of course, the user may still mutate the state otherwise (either through the useNavigation hook, direct access to the Reveal camera manager, or just moving the camera with mouse or keyboard), and this will be handled gracefully by the RevealContext (the setter sent to RevealContext will be called in all these cases)

## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [ ] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [ ] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [ ] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [ ] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [ ] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [ ] I have noted down and am currently tracking any technical debt introduced in this PR.
- [ ] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: react-components/src/components/RevealCanvas/ViewerContext.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+export const ViewerContextProvider = ({
+  cameraState,
+  setCameraState,
+  value,
+  children
+}: {
+  cameraState?: CameraStateParameters;
+  setCameraState?: (cameraState?: CameraStateParameters) => void;
+  value: RevealRenderTarget | null;
+  children: ReactNode;
```
Comment (pramodcog) [inline react-components/src/components/RevealCanvas/ViewerContext.tsx:20]: Can be moved to `props`.

File: react-components/src/components/RevealCanvas/ViewerContext.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  setCameraState
+}: {
+  cameraState?: CameraStateParameters;
+  setCameraState?: (cameraState?: CameraStateParameters) => void;
+}): ReactNode => {
+  useCameraStateControl(cameraState, setCameraState);
+  return null;
+};
+
+export const useReveal = (): Cognite3DViewer => {
```
Comment (pramodcog) [inline react-components/src/components/RevealCanvas/ViewerContext.tsx:41]: This should be in `hooks` folder?

File: react-components/src/components/RevealCanvas/hooks/useCameraStateControl.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+
+import { type MutableRefObject, useEffect, useRef } from 'react';
+import { useReveal } from '../../..';
```
Comment (pramodcog) [inline react-components/src/components/RevealCanvas/hooks/useCameraStateControl.ts:6]: Proper path!

**Review by pramodcog — APPROVED** [review]
Looks good, nice works with the tests!
Minor comments

File: react-components/src/components/RevealCanvas/ViewerContext.tsx (hunk 2 above)
Comment (haakonflatval-cognite) [reply react-components/src/components/RevealCanvas/ViewerContext.tsx:41]: I don't think so, the convention we have been using so far, is that hooks that are sort-of exposing values from a Context, are put together with that Context. That's how this hook was placed before this PR as well, in ViewerContext.ts

**Pull request merged**